src/oaklib/interfaces/differ_interface.py

```diff
@@ -66,125 +91,240 @@ def diff(
+                        yield ClassCreation(id=_gen_id(), about_node=node)
+                    else:
+                        class_creations.append(ClassCreation(id=_gen_id(), about_node=node))
                 else:
-                    logging.info(f"Both ontologies have no definition for {e1}")
-
-            differs = self.different_from(e1, other_ontology)
-            if differs is not None and not differs:
-                continue
-            e1_arels = set(self.alias_relationships(e1, exclude_labels=True))
-            e2_arels = set(other_ontology.alias_relationships(e1, exclude_labels=True))
-            for arel in e1_arels.difference(e2_arels):
-                pred, alias = arel
-                switches = {r[0] for r in e2_arels if r[1] == alias}
-                if len(switches) == 1:
-                    e2_arels = set([x for x in e2_arels if x[1] != alias])
-                    # TODO: KGCL model needs to include predicates
-                    yield kgcl.SynonymPredicateChange(id=_gen_id(), about_node=e1, old_value=alias)
+                    if configuration.yield_individual_changes:
+                        yield NodeCreation(id=_gen_id(), about_node=node)
+                    else:
+                        node_creations.append(NodeCreation(id=_gen_id(), about_node=node))
+
+            # Now yield based on the flag and whether there are items in the lists
+            if not configuration.yield_individual_changes:
+                changes = {}
+                if class_creations:
+                    changes[CLASS_CREATION] = class_creations
+                if node_creations:
+                    changes[NODE_CREATION] = node_creations
+                if changes:
+                    yield changes
+
+        # Node Deletion with consideration for yield_individual_changes flag
+        nodes_to_delete = self_entities - other_ontology_entities
+
+        if configuration.yield_individual_changes:
+            for node in nodes_to_delete:
+                yield NodeDeletion(id=_gen_id(), about_node=node)
+        else:
+            yield {
+                NODE_DELETION: [
+                    NodeDeletion(id=_gen_id(), about_node=node) for node in nodes_to_delete
+                ]
+            }
+
+        # ! Obsoletions
+        other_ontology_entities_with_obsoletes = set(
+            other_ontology.entities(filter_obsoletes=False)
+        )
+        other_ontology_entities_without_obsoletes = set(
+            other_ontology.entities(filter_obsoletes=True)
+        )
+        other_ontology_obsolete_entities = (
+            other_ontology_entities_with_obsoletes - other_ontology_entities_without_obsoletes
+        )
+        possible_obsoletes = self_entities.intersection(other_ontology_obsolete_entities)
+        obsoletion_generator = _generate_obsoletion_changes(
+            possible_obsoletes,
+            self.entity_metadata_map,
+            other_ontology.entity_metadata_map,
+        )
```

Comment:
New strategy to get `obsoleted` nodes (lines 167 - 181). @cmungall , thoughts? This cuts down runtime from 16 mins (original code) to 2 mins. The results seem similar. I want to confirm if the logic is sound and if I'm not missing anything.

cc: @matentzn